**Titanic dataset**

Research question: Were the wommen and children among three different Pclasses on Titanic be treated equally (same priority to get on a lifeboat)? 

hypotheses: Women and children in third class were not treated equally as those in first or second class.

We all know that women and children on Titanic had more chance to survive since women and children were given priority to use life boat. However I doubt that women and children among three different Pclasses were treated equally, i.e. they got almost the same chance to survive.

In [59]:
#import Pandas and Numpy package
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

#read titanic-data.csv file, storing in pandas dataframe
df = pd.read_csv('titanic-data.csv', header=0)

#look at how is the dataset look like
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
PassengerId    891 non-null int64
Survived       891 non-null int64
Pclass         891 non-null int64
Name           891 non-null object
Sex            891 non-null object
Age            714 non-null float64
SibSp          891 non-null int64
Parch          891 non-null int64
Ticket         891 non-null object
Fare           891 non-null float64
Cabin          204 non-null object
Embarked       889 non-null object
dtypes: float64(2), int64(5), object(5)
memory usage: 83.6+ KB


Now we have imported the packages and read in the CSV file. Let's look at the general information of the data. The dataset includes 891 observations of 12 variables. Some variables (columns) contains missing values. For example, there are 177 missing values in "Age" column. 

In [60]:
#Look at the overall survival situation of female among three Pclasses
df_female = df[df['Sex']=='female'][['Sex', 'Pclass', 'Survived']]
df_female


Sex  Pclass  Survived
1    female       1         1
2    female       3         1
3    female       1         1
8    female       3         1
9    female       2         1
10   female       3         1
11   female       1         1
14   female       3         0
15   female       2         1
18   female       3         0
19   female       3         1
22   female       3         1
24   female       3         0
25   female       3         1
28   female       3         1
31   female       1         1
32   female       3         1
38   female       3         0
39   female       3         1
40   female       3         0
41   female       2         0
43   female       2         1
44   female       3         1
47   female       3         1
49   female       3         0
52   female       1         1
53   female       2         1
56   female       2         1
58   female       2         1
61   female       1         1
..      ...     ...       ...
807  female       3         0
809  female       1         1
813  female       3         0
816  female       3         0
820  female       1         1
823  female       3         1
829  female       1         1
830  female       3         1
835  female       1         1
842  female       1         1
849  female       1         1
852  female       3         0
853  female       1         1
854  female       2         0
855  female       3         1
856  female       1         1
858  female       3         1
862  female       1         1
863  female       3         0
865  female       2         1
866  female       2         1
871  female       1         1
874  female       2         1
875  female       3         1
879  female       1         1
880  female       2         1
882  female       3         0
885  female       3         0
887  female       1         1
888  female       3         0

[314 rows x 3 columns]

Then we create a new dataframe object (df_female) to store all the female obsevations which only includes three variables, "Sex", "Pclass", "Survived".

After looking at the overall survival situation of female among three classes above, it seems the results may meet my expectation. Now I want to dive into the observations of each Pclass separately.

In [61]:
#look at the survival situation of female in first class
df_female_first_class = df[(df['Sex']=='female') & (df['Pclass']==1)][['Sex', 'Pclass', 'Survived']]
df_female_first_class['Survived'].describe()

count    94.000000
mean      0.968085
std       0.176716
min       0.000000
25%       1.000000
50%       1.000000
75%       1.000000
max       1.000000
Name: Survived, dtype: float64

We create a new dataframe object (df_female_first_class) to store all the female observations from "first class". we can see that there 94 females in first class. 

In [62]:
#display as bar chart 
df_female_first_class['Survived'].value_counts().plot(kind= 'bar')







From the bar chart above we can see that most females in first class survived.

In [63]:
##look at the survival situation of female in second class
df_female_second_class = df[(df['Sex']=='female') & (df['Pclass']==2)][['Sex', 'Pclass', 'Survived']]
df_female_second_class['Survived'].describe()

count    76.000000
mean      0.921053
std       0.271448
min       0.000000
25%       1.000000
50%       1.000000
75%       1.000000
max       1.000000
Name: Survived, dtype: float64

We create a new dataframe object (df_female_second_class) to store all the female observations from "second class". we can see that there 76 females in second class. 

In [64]:
#display as bar chart
df_female_second_class['Survived'].value_counts().plot(kind = 'bar')

From the bar chart above we can see that most females in second class survived.

In [65]:
##look at the survival situation of female in third class
df_female_third_class = df[(df['Sex']=='female') & (df['Pclass']==3)][['Sex', 'Pclass', 'Survived']]
df_female_third_class['Survived'].describe()

count    144.000000
mean       0.500000
std        0.501745
min        0.000000
25%        0.000000
50%        0.500000
75%        1.000000
max        1.000000
Name: Survived, dtype: float64

We create a new dataframe object (df_female_third_class) to store all the female observations from "third class". we can see that there 144 females in third class. 

In [66]:
#display as bar chart
df_female_third_class['Survived'].value_counts().plot(kind = 'bar', title = 'Survival situation of third-class women')

From the bar chart above we can see that only half of the females in third class survived.

From the data analysis above, the survival rate of female in first Pclass is 96.8%; the survival rate of female in second Pclass is 92.1%；the survival rate of female in third Pclass is 50%. There is significant difference between them. But do this mean that wealthy were put on the lifeboat first? Not necessarily! Maybe the cabins of third class were at more bottom of Titanic, so females of third class were more easily flooded by water and needed more time to get to the deck. Since 687 values in "Cabin" column are missing, we'll stop here.

Now let's investigate the situation of children. We assume children's age are between 0 to 14.

Since we need accurate "Age" value to avoid bias in this investigation, we'll drop all the observations without age information.  

In [67]:
# drop the data without age information
df_first_class = df[df['Pclass']==1][['Age', 'Survived']].dropna()

#look at the survival situation of chidren(age:0-14) in first class
df_children_first_class = df_first_class[df_first_class['Age'] < 14.0][['Age', 'Survived']]
df_children_first_class.describe()

Age  Survived
count   4.000000      4.00
mean    4.480000      0.75
std     4.530077      0.50
min     0.920000      0.00
25%     1.730000      0.75
50%     3.000000      1.00
75%     5.750000      1.00
max    11.000000      1.00

After dropping observations of first class that without age information, we can see there are 4 children in first class. And the mean of variable "Survived" is '0.75'.

In [68]:
#display as pie chart
df_children_first_class['Survived'].value_counts().plot(kind='pie')

From the pie chart, we can see the rate of survial of first-class children is 75%. However, the size is too small, so the results may be largely biased.

In [69]:
df_second_class = df[df['Pclass']==2][['Age','Survived']].dropna()

#look at the survival situation of chidren(age:0-14) in second class
df_children_second_class = df_second_class[df_second_class['Age'] < 14.0][['Age', 'Survived']]
df_children_second_class.describe()

Age  Survived
count  18.000000      18.0
mean    4.018333       1.0
std     3.318226       0.0
min     0.670000       1.0
25%     1.250000       1.0
50%     3.000000       1.0
75%     5.750000       1.0
max    13.000000       1.0

After dropping observations of second class that without age information, we can see there are 18 children in second class. And the mean of variable "Survived" is '1'.

In [70]:
#display as pie chart
df_children_second_class['Survived'].value_counts().plot(kind='pie')

From the pie chart, we can see the rate of survial of second-class children is 100%. All of the children in this class survived!

In [71]:
df_third_class = df[df['Pclass']==3][['Age','Survived']].dropna()

#look at the survival situation of chidren(age:0-14) in third class
df_children_third_class = df_third_class[df_third_class['Age'] < 14.0][['Age', 'Survived']]
df_children_third_class.describe()

Age   Survived
count  49.000000  49.000000
mean    5.406531   0.428571
std     3.645783   0.500000
min     0.420000   0.000000
25%     2.000000   0.000000
50%     4.000000   0.000000
75%     9.000000   1.000000
max    13.000000   1.000000

After dropping observations of third class that without age information, we can see there are 49 children in third class. And the mean of variable "Survived" is less than '0.5'.

In [72]:
#display as pie chart
df_children_third_class['Survived'].value_counts().plot(kind='pie')


From the pie chart, we can see that less than half of the children in third class survived.

As we can see above, the survival rate of children in third class was significantly lower than those in first and second class. However, is it safe to say that children in third class were not treated equally as those in first or second class? No, I was wondering that maybe children in third class had more siblings and because of that, their parents had more difficulty to take care of all their children. In contrast, maybe children in first or second class had fewer siblings, so it is easier for their parents to take care of them. Let's figure it out!!

In [73]:
#create a new dataframe which includes four variables from original dataset, and drop the observations without age 
#information
df_new = df[['Age', 'Pclass', 'SibSp', 'Survived']].dropna()
df_new.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 714 entries, 0 to 890
Data columns (total 4 columns):
Age         714 non-null float64
Pclass      714 non-null int64
SibSp       714 non-null int64
Survived    714 non-null int64
dtypes: float64(1), int64(3)
memory usage: 27.9 KB


Now we hava a new dataset including 4 columns and 714 observations without NA values. Also, we assume all the children under 14 have no spouse. Therefor the 'SibSp' value is equal to the number of each child's siblings. 

In [74]:
#create an object to store 'SibSp' value of each child in first class
df_sib_first_class_children = df_new[(df_new['Pclass'] ==1) & (df_new['Age'] < 14.0)]['SibSp']
df_sib_first_class_children.describe()

count    4.00
mean     0.75
std      0.50
min      0.00
25%      0.75
50%      1.00
75%      1.00
max      1.00
Name: SibSp, dtype: float64

As the statistic show, on average, each child in second class had fewer than 1 sibling.

In [75]:
#create an object to store 'SibSp' value of each child in third class
df_sib_third_class_children = df_new[(df_new['Pclass'] ==3) & (df_new['Age'] < 14.0)]['SibSp']
df_sib_third_class_children.describe()

count    49.000000
mean      2.285714
std       1.744037
min       0.000000
25%       1.000000
50%       3.000000
75%       4.000000
max       5.000000
Name: SibSp, dtype: float64

Comparing to first and second class, the children in third class often had more siblings on Titanic. On average they had more than two siblings. This maybe one of the factors that influenced the relatively low survial rate of third-class children.

**Conclusion**:

I started to explore the dataset with a hypotheses that women and children in third class were not treated equally as those in first or second class. Fisrt, I investigated the survial rate of women in each class and found the surival rate of third-class (50.0%) women were significant lower than those in first (96.8%) and second class (92.1%). Then I investigated the survial rate of children (age < 14) in each class and the data told us the survival rate of third-class children (42.9%) were significant lower than those in first (75%) and second class (100%). 

However, it is not safe to accept our hypotheses, becasue there might be other factors that casue such results. One guess is the cabins of third class were at more bottom of Titanic, so peopel of third class were more easily flooded by water and needed more time to get to the deck. Since 687 values in "Cabin" column are missing, I stoped further investigation. Another guess is that maybe children in third class had more siblings and because of that, their parents had more difficulty to take care of all their children. So I did further investigation from this perspective and found that on average, each child in third class had more than two siblings while child in other classed only had less than one sibling. So it is possible the number of children's sibling is a factor that influences children's survial rate.

There is also limitation in this study. For example, although I investigated the number of children's sliblings, but I did not investigate the number of children's parents. Although third-class children had more siblings, but we don't know if they also had more parents on Titanic. Also, based on this dataset, we cannot get the infomation about the sex of children's parents. Having father or mother or both on Titanic may make a difference to children's survival rate.

reference:

1.http://stackoverflow.com/questions/31029560/plotting-categorical-data-with-pandas-and-matplotlib